### DISCUSSION TITLE
Is there a way to disable selections when doing motions?

### DISCUSSION THREAD

**Discussion by art-solopov** [meow-edit/meow#299] [Q&A]
When I do a movement in Meow (for example, `] l` to jump at the end of the line) I not only do the motion but also have a selection from my current point to wherever I move. I have to disable the selection manually. Is there a way to disable this behaviour? Thank you.

**Comment by Linerre** [comment]
To jump to the end of a line, `C-e` works out very well. Wondering any specific reason you prefer to do the jumping with Meow? If you are from Vim-like editing styles, then I can imagine your surprise/confusion. 

My understanding is that one key idea behind Meow is that selection is tightly bound with movement. 

`] l` not only moves the cursor to the line end (as `C-e` does), it also selects the path (in characters) it passes through so that kill/yank is just one or two keys away.

**Reply by trev-dev** [reply]
More to your point, if you're using Meow to get around and are trying to *minimize* your control-key modifiers, it becomes less of a pain for your hands. 

My examples below follow the example set-up for Qwerty.

Instead of `C-e` try just pressing `meow-line` (x). You're instantly on the end of the line.

Want to start tying here? Press `meow-append` (a). Did you want to type at the front of the line? Press `meow-insert` (i).

You can yank text from your kill ring anywhere the cursor is at rest without worrying about your selection. Perhaps you want to quickly yank at the front of the line? Just go `x;y` to use `meow-reverse` before yanking.

All of the tips mentioned so far apply to _any kind of selection_ you might make in meow.

If the selection is so unsightly for you, just press `g` and cancel the selection. However, with Meow, the selection has more utility than just setting a region to manipulate or kill. Meow has it's own mark history that you can easily pop with `meow-pop-history` (z) if you make a selection error such as over-shooting your target. Embracing "selections first" lets you take advantage of this. 

Moreover I would argue that the _visual_ aspect of following the selection is useful feedback to help you appreciate which direction you're moving in and where your cursor is, moment-to-moment.

**Reply by trev-dev** [reply]
Oh and one more tip. Try removing any `:foreground` color to your `'region` or `'secondary-selection` faces. For myself this makes using "selections to travel" much less disruptive as I get to see my syntax highlighting.

**Comment by eshrh** [comment]
The above answer is quite right i think. Meow movement and selections are inherently tied together, in fact every movement except for simple left-right-up-down will create a selection. It certainly is possible to hack the source and get the results you want, but it would go against the general philosophy of the project.

This style of "always in visual mode" is also in kakoune, and IMO it's one of the biggest advantages compared to *vim once you get used to it.
